pkg/fleets/controller.go

```diff
@@ -461,18 +477,53 @@ func (c *Controller) rollingUpdateActive(fleet *agonesv1.Fleet, active *agonesv1
 }
 
 // rollingUpdateRest applies the rolling update to the inactive GameServerSets
-func (c *Controller) rollingUpdateRest(fleet *agonesv1.Fleet, rest []*agonesv1.GameServerSet) error {
+func (c *Controller) rollingUpdateRest(fleet *agonesv1.Fleet, active *agonesv1.GameServerSet, rest []*agonesv1.GameServerSet) error {
 	if len(rest) == 0 {
 		return nil
 	}
 
-	r, err := intstr.GetValueFromIntOrPercent(fleet.Spec.Strategy.RollingUpdate.MaxUnavailable, int(fleet.Spec.Replicas), true)
+	// Look at Kubernetes Deployment util ResolveFenceposts() function
+	var r int
+	var err error
+	if runtime.FeatureEnabled(runtime.FeatureRollingUpdateOnReady) {
+		r, err = intstr.GetValueFromIntOrPercent(fleet.Spec.Strategy.RollingUpdate.MaxUnavailable, int(fleet.Spec.Replicas), false)
```

Comment:
I suggest checking error here in order to avoid all those steps below including type casts and so on. What do you think?

Reply:
I assume the error is going to be rare? (or close to impossible?), in which case I'm happy with this code as is.

Reply:
Well let me add an error here.